### PR TITLE
Define settings to change django-static-compress behaviors.

### PR DESCRIPTION
**Pull request by hongquan**
Settings to define:

- `STATIC_COMPRESS_FILE_EXTS`
- `STATIC_COMPRESS_METHODS`
- `STATIC_COMPRESS_KEEP_ORIGINAL`
- `STATIC_COMPRESS_MIN_SIZE_KB`

### CODE REVIEW

**Comment by hongquan** [comment]
@whs Could you take a look?

**Comment by whs** [comment]
Could you add some tests. For example:

- Test that if STATIC_COMPRESS_FILE_EXTS is honored if changed.
- Test that STATIC_COMPRESS_FILE_EXTS can be empty
- Test STATIC_COMPRESS_KEEP_ORIGINAL=False is working

**Comment by hongquan** [comment]
Hi @whs, I added test cases and added `gz+zop`.

**Comment by hongquan** [comment]
Fixed my bug with `ZlipCompressor`.

File: static_compress/mixin.py [hunk 1]
```diff
@@ -1,11 +1,73 @@
...
 from . import compressors
 
 __all__ = ['CompressMixin']
 
 
+DEFAULT_METHODS = ['gz', 'br']
+METHOD_MAPPING = {
+	'gz': compressors.ZlibCompressor,
+	'br': compressors.BrotliCompressor,
+	'gz+zop': compressors.ZopfliCompressor,
```
Comment (whs) [inline static_compress/mixin.py]: I prefer `gz+zopfli` so it is Google-able if someone found this in the settings without knowing what it is

File: static_compress/mixin.py [hunk 2]
```diff
@@ -1,11 +1,73 @@
...
+from os.path import getatime, getctime, getmtime
+
+from django.core.exceptions import ImproperlyConfigured
+
 from . import compressors
 
 __all__ = ['CompressMixin']
 
 
+DEFAULT_METHODS = ['gz', 'br']
```
Comment (whs) [inline static_compress/mixin.py]: The default should be `gz+zop` so it won't break existing users.

Plus, this module is all about trading build speed for filesize, so we should prefer the best options first.

Comment (hongquan) [reply static_compress/mixin.py]: Hi, it won't break, because it still produce gzip files, just with bigger files.

Comment (whs) [reply static_compress/mixin.py]: While it won't break apps, it would break the user's expectation

**Comment by hongquan** [comment]
Hi, is a test case for `STATIC_COMPRESS_MIN_SIZE_KB` really needed? Because all test cases already change `STATIC_COMPRESS_MIN_SIZE_KB` to make _django-static-compress_ work with small files included (which is from <1kB to 4kB) as testing data. In one test case, I also checked if the file smaller than `STATIC_COMPRESS_MIN_SIZE_KB` is not compressed.

**Comment by whs** [comment]
I see the checks now, thanks for pointing me.

**Comment by hongquan** [comment]
Hi, I made Zopfli default for `gz`, and define `gz+zlib` for `zlib` compressor.

**Comment by whs** [comment]
Thanks for your pull request.

I'll cut a release by tomorrow. Feel free to ping me if I haven't do that by the next week.

**Comment by whs** [comment]
v1.2.0 has been published :tada: 

https://pypi.org/project/django-static-compress/

**Pull request merged**